lib/layer.js

```diff
@@ -48,6 +48,31 @@ function Layer(path, options, fn) {
 }
 
 /**
+ * Handle any request for the layer.
+ *
+ * @api private
+ */
+
+Layer.prototype._handle_any = function _handle_any(/* ...args, next */) {
+  var self = this
+  var next = arguments[arguments.length - 1]
+  var handle = this.handle
+  var result
+  var hasNextBeenCalled = function () { return next._currentLayer === self }
+
+  try {
+    result = handle.apply(undefined, arguments)
```

Comment:
Prior to this change, the `handle` was called with the default context (`global` in sloppy mode and `null` in strict mode). Now this is being changed to always provide a context of `undefined`. This will probably break some middleware. Was there a reason to change the context of the handle call? Can we search `npm` to determine the swath of modules this change would effect?